docs/testplan/PFC_Congestion_Oversubscription_Test_Plan.md

```diff
@@ -0,0 +1,318 @@
+  *	The test data traffic will consist of two lossless traffic streams, with the SONiC default lossless priorities of 3 and 4, and each having a 25% bandwidth.
+  * PFC pause storm: Persistent PFC pause frames from the IXIA Rx port. The priority of PFC pause frames will be randomized to those of the test data traffic streams, or include all lossless priorities. The inter-frame transmission interval should be smaller than the per-frame pause duration.
+The mapping of background and test data traffic streams to IXIA transmit ports can be randomized while keeping two streams per port; for example, one test port could send two lossless traffic streams and the second port two lossy streams, or each port could send two mixed streams, one lossless, one lossy.
+
+This experiment needs the following six steps:
+1.	Start the background and test traffic from IXIA Tx ports 1 & 2; the combined line rate will be 100% of the egress port capacity (25% line rate bandwidth per stream)
+2.	After a few seconds, start the PFC pause storm to fully block one or all test priorities at the switch. Ensure that:
+a.	The IXIA Rx port is receiving all lossy streams and any of the lossless streams not marked for PFC pause due to randomization.
+b.	The IXIA Rx port does not receive any of the lossless streams marked to be fully blocked by PFC.
+c.	If possible, query the DUT drop and watermark counters
+3.	After a few more seconds, stop the PFC pause storm; ensure that the IXIA Rx port statistics show that 100% of the traffic is being received.
+4.	Repeat steps 2 & 3 a few times, possibly with different PFC storm durations.
+5.	Stop the PFC storm and all traffic.
+6.	On the IXIA Rx port ensure that:
+a.	The received lossless traffic streams on IXIA port 3 show no missing and no out-of-order frames.
+b.	The lossy traffic streams did not drop frames.
+
+### Test case # 6 – Lossless Response to Throttling PFC Pause
+#### Test objective
+Test the reduction in lossless streams bandwidth when subject to throttling PFC pause messages. The PFC watchdog, which should be enabled for this test, should not intervene.
+
+#### Background
+This test uses the same configuration and test bed as  similar diagram and description of the multi-ASIC router as the ‘Lossless Response to External PFC Pause Storms ' test case.
+
+In this test case we will also assume that all the lossy and lossless traffic priorities have a WRR scheduler configured with the same weights across all eight priorities, with no strict priority (SP) priorities configured.
+
+#### Testbed Setup
+The test bed setup consists of three IXIA ports and a SONiC router as the device under test (DUT). All IXIA ports should have the same line speed.  The three DUT ports will be chosen at random across all the available ports in the router in order to eventually cover combinations of ports within the same line card and across line cards (or ports within the same ASIC vs. different ASICs on a single-card, multi-ASIC design).

```

Comment:
Can we test mix port speeds here ?

Reply:
We can use different port speeds. But to validate the expected outcome, we are using same speed modes. If not it will be very difficult to meet the expectations for different speed modes.